examples/data-objects/inventory-app/src/inventoryList.ts

```diff
@@ -4,31 +4,58 @@
  */
 
 import { DataObject, DataObjectFactory } from "@fluidframework/aqueduct";
-import { ISharedTree, SharedTreeFactory } from "@fluid-experimental/tree2";
+import {
+	AllowedUpdateType,
+	ForestType,
+	TypedTreeChannel,
+	TypedTreeFactory,
+	typeboxValidator,
+} from "@fluid-experimental/tree2";
 import { IFluidHandle } from "@fluidframework/core-interfaces";
+import { InventoryField, inventoryField, schema } from "./schema";
 
 const treeKey = "tree";
 
+const factory = new TypedTreeFactory({
+	jsonValidator: typeboxValidator,
+	forest: ForestType.Reference,
+	initialTree: {
+		parts: [
+			{
+				name: "nut",
+				quantity: 0,
+			},
+			{
+				name: "bolt",
+				quantity: 0,
+			},
+		],
+	},
+	allowedSchemaModifications: AllowedUpdateType.None,
+	schema,
+	subtype: "InventoryList",
+});
+
 export class InventoryList extends DataObject {
-	private _tree: ISharedTree | undefined;
+	private _tree: TypedTreeChannel<typeof inventoryField> | undefined;
 
-	public get tree(): ISharedTree {
+	public get tree(): InventoryField {
 		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-		return this._tree!;
+		return this._tree!.root;
 	}
 
 	protected async initializingFirstTime() {
-		this._tree = this.runtime.createChannel(
-			undefined,
-			new SharedTreeFactory().type,
-		) as ISharedTree;
-
-		this.root.set(treeKey, this.tree.handle);
+		this._tree = this.runtime.createChannel(undefined, factory.type) as TypedTreeChannel<
+			typeof inventoryField
+		>;
+		this.root.set(treeKey, this._tree.handle);
 	}
 
 	protected async initializingFromExisting() {
 		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-		this._tree = await this.root.get<IFluidHandle<ISharedTree>>(treeKey)!.get();
+		this._tree = await this.root
```

Comment:
I think I found a ST bug while trying a merge with my PR.  I had been using `hasInitialized` to retrieve/use the tree from the handle regardless of whether this was the initial creation or from existing.  However, on creation the tree I retrieve from the handle has no `root` member!  If you change this method from `initializingFromExisting` to `hasInitialized` I'm guessing you'd be able to repro.

Reply:
Sounds like you may have hit the bug fixed in https://github.com/microsoft/FluidFramework/pull/17594

Reply:
Nah I just merged main like a half hour ago - are you able to repro?